### PR TITLE
CI : Add GitHub action

### PR DESCRIPTION
**Pull request by zekth**
Duration of the matrix build is around 2minutes.
Example : https://github.com/zekth/pino/commit/1bd66552bd64c78585c83596ad11829be3eae497/checks?check_suite_id=314800692

So i've simply moved everything to Github action CI. I see there is a secure env var in the `travis.yml` but the CI is not failing without it; is it not used anymore in the CI process? Otherwise we can use `Secrets` vars in the project setting : https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets

Also do i delete the `travis.yml` ?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+on: [push, pull_request]
+jobs:
+  build:
+    name: ${{ matrix.node-version }} ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-16.04]
+        node-version: [6, 8, 10, 11]
```
Comment (mcollina) [inline .github/workflows/ci.yml]: Would you mind adding 12 and 13?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: CI
+on: [push, pull_request]
+jobs:
+  build:
+    name: ${{ matrix.node-version }} ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-16.04]
```
Comment (mcollina) [inline .github/workflows/ci.yml]: Would you mind using ubuntu-latest, and add WIndows and Mac OS X as well?

**Comment by zekth** [comment]
I've added node 12 / 13 and win and mac os. Let's see the result of the run : https://github.com/zekth/pino/commit/f82147bcd4f6b2a40805c123d77ff9a53eb188cc/checks?check_suite_id=314805922

**Comment by mcollina** [comment]
> For win CI the threshold may need to be adjusted

Please proceed.

> `# Subtest: http request support via serializer without request connection`

This is due to a change in Node core. We have to change this to have `is` assertions for specific values. Would you mind to take care of that change?

**Comment by mcollina** [comment]
> This is due to a change in Node core. We have to change this to have is assertions for specific values. Would you mind to take care of that change?

I implied only for that specific test :). SOrry for the confusion.

**Comment by zekth** [comment]
Everything is green now :)
https://github.com/zekth/pino/commit/4726199ca7ed564d4ab63573c0ef108a160dcb3a/checks?check_suite_id=315896709

**Comment by zekth** [comment]
> > This is due to a change in Node core. We have to change this to have is assertions for specific values. Would you mind to take care of that change?
> 
> I implied only for that specific test :). SOrry for the confusion.

Yeah moved to `isLike`

**Comment by zekth** [comment]
We can set ubuntu as referall for coverall don't you think?

**Comment by jsumners** [comment]
Yes.

**Comment by zekth** [comment]
I'll do it in a further PR. Do we keep travis for the transition period?

**Comment by jsumners** [comment]
Sounds like a reasonable thing to do.

**Comment by zekth** [comment]
Seems like the tests for windows are not stable : https://github.com/pinojs/pino/runs/307985680

I'll try to find out why

**Pull request merged**